geode-redis/src/main/java/org/apache/geode/redis/internal/executor/set/RedisSet.java

```diff
@@ -34,18 +33,15 @@
 import org.apache.geode.DataSerializer;
 import org.apache.geode.Delta;
 import org.apache.geode.InvalidDeltaException;
-import org.apache.geode.cache.Region;
 import org.apache.geode.redis.internal.ByteArrayWrapper;
 import org.apache.geode.redis.internal.Coder;
 
 /**
- * This class still uses "synchronized" to protect the
- * underlying HashSet even though all writers do so under
- * the {@link SynchronizedStripedExecutor}. The synchronization on this
- * class can be removed once readers are changed to
- * also use the {@link SynchronizedStripedExecutor}.
+ * This class still uses "synchronized" to protect the underlying HashSet even though all writers do
+ * so under the {@link SynchronizedStripedExecutor}. The synchronization on this class can be
+ * removed once readers are changed to also use the {@link SynchronizedStripedExecutor}.
  */
-public class RedisSet implements Delta, DataSerializable {
+public class RedisSet implements Delta, DataSerializable, RedisSetCommands {
```

Comment:
I don't think you should force RedisSetCommands on this class. None of the methods on this class currently need the "key" parameter because they do not need to find themselves in the region. But RedisSetCommands methods do  need a "key". I think that abstraction really does not fit this class.

Reply:
yea, i may have been too ambitious in trying to get all the layers to conform to this interface...i think it's a good long-term goal to have, but we still need some changes to make this possible..As long as some commands go through the executor and some take a "side" route, it will be difficult to get this pattern implemented.